src/Core/src/Platform/iOS/AcceleratorExtensions.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Linq;
+using Foundation;
+using ObjCRuntime;
+using UIKit;
+
+namespace Microsoft.Maui.Platform
+{
+	internal static class AcceleratorExtensions
+	{
+		internal static UIMenuElement CreateMenuItem(this IMenuFlyoutItem virtualView, IMauiContext mauiContext)
+		{
+			int index = MenuFlyoutItemHandler.menus.Count;
+			UIImage? uiImage = virtualView.Source.GetPlatformMenuImage(mauiContext);
+			var selector = new Selector($"MenuItem{index}:");
+
+			bool selectorFound =
+				MauiUIApplicationDelegate.Current.RespondsToSelector(selector);
+
+			if (!selectorFound)
+			{
+				throw new InvalidOperationException(
+					$"By default we only support 50 MenuItems. You can add more by adding the following code to {MauiUIApplicationDelegate.Current.GetType()}\n\n" +
+					$"[Export(\"MenuItem{index}: \")]\n" +
+					$"internal void MenuItem{index}(UICommand uICommand)\n" +
+					"{\n" +
+					"	uICommand.SendClicked();\n" +
+					"}");
+			}
+
+			var accelerator = virtualView.Accelerators?[0];
+			if (accelerator is null)
+				return virtualView.CreateMenuItemCommand(index, uiImage, selector);
+
+			var key = accelerator.Key;
+			var modifiers = accelerator.Modifiers;
+
+			UIKeyModifierFlags modifierFlags = 0;
+
+			// Single key accelerators (no modifier, i.e. "A") and multi-key accelerators are supported (>=1 modifier and 1 key only, i.e. Ctrl+F, Ctrl+Shift+F)
+			if (modifiers is not null)
+			{
+				var modifiersCount = modifiers.Count();
+
+				if (modifiersCount > 0)
+				{
+					for (int i = 0; i < modifiersCount; i++)
+					{
+						var modifierMask = modifiers.ElementAt(i).ToLowerInvariant();
+
+						switch (modifierMask)
+						{
+							case "ctrl":
+								modifierFlags |= UIKeyModifierFlags.Control;
+								break;
+							case "cmd":
+								modifierFlags |= UIKeyModifierFlags.Command;
+								break;
+							case "alt":
+								modifierFlags |= UIKeyModifierFlags.Alternate;
+								break;
+							case "shift":
+								modifierFlags |= UIKeyModifierFlags.Shift;
+								break;
+						}
+					}
+				}
+			}
+
+			return virtualView.CreateMenuItemKeyCommand(index, uiImage, selector, modifierFlags, key);
+		}
+
+		static UIMenuElement CreateMenuItemKeyCommand(this IMenuFlyoutItem virtualView, int index, UIImage? uiImage, Selector selector, UIKeyModifierFlags modifierFlags, string key)
+		{
+			var keyCommand = UIKeyCommand.Create(
+										title: virtualView.Text,
+										uiImage,
+										selector,
+										key,
+										modifierFlags,
+										new NSString($"{index}"));
```

Comment:
This is what the compiler will generate for $"{index}" vs ToString:

Source:
```csharp
using System;
public class Window {
    
    public static string Test(int index) {
        return $"{index}";
    }
    
    public static string Test1(int index) {
        return index.ToString();
    }
}
```

Generated code:
```csharp
[System.Runtime.CompilerServices.NullableContext(1)]
[System.Runtime.CompilerServices.Nullable(0)]
public class Window
{
    public static string Test(int index)
    {
        DefaultInterpolatedStringHandler defaultInterpolatedStringHandler = new DefaultInterpolatedStringHandler(0, 1);
        defaultInterpolatedStringHandler.AppendFormatted(index);
        return defaultInterpolatedStringHandler.ToStringAndClear();
    }

    public static string Test1(int index)
    {
        return index.ToString();
    }
}
```

Use to int.ToString() method rather than an interpolation, $"{index}" Add unnecessary code when we simply want to use the string value.

Reply:
Nice feedback. Changed.